05/ukol7.py

```diff
@@ -0,0 +1,57 @@
+#Napiš tyto programy. Každý z nich bude mít na vstupu řetězec s rodným číslem a nějak ho
+#zanalyzuje:
+#(a) Je ve správném formátu: 6 číslice, lomítko, 4 číslice? (vypíše True nebo False)
+#(b) Je dělitelné jedenácti? (vypíše True nebo False)
+#(c) Jaké datum narození je v čísle zakódováno? (vypíše trojici čísel – den, měsíc, rok)
+#(d) Jaké pohlaví je v čísle zakódováno? (vypíše 'muž' nebo 'žena')
+#Pro účely úkolu stačí, když bude program umět zpracovat čísla vydávaná od roku 1985.
+#Reálná rodná čísla můžou být složitější :)
+
+
+
+#Osetreni vstupu
+while True:
+    pin = input('Zadej rodne cislo ve formatu xxxxxx/xxxx: ') 
+    if pin.replace('/','',1).isdigit():
+        break
+    print('Toto neni cislo ve formatu xxxxxx/xxxx!')
+
+#Pin musi mit 11 znaku, na seste pozici musi byt /, kdyz se lomitko na prvni pozici nahradi "", tak musi byt pin cislo.
+if len(pin) == 11 and pin[6] == '/' and pin.replace('/','',1).isdigit():
+    print('Kontrola formatu cisla: ', True)
+else:
+    print('Kontrola formatu cisla: ', False) 
+    exit(1)   
+
+pin = pin.replace('/','',1)         #Prvni pozici se znakem / vymenim za ""
+pin_int = int(pin)                  #Rodne cislo prevedu ze string na cislo
+result = pin_int / 11               #Rodne cislo vydelim 11 se zbytem
+
+if result == int(result):           #Vysledek musi byt cele cislo
+    #print(result)
+    print('Delitelnost 11: ', True)
+else:
+    print('Delitelnost 11: ',False) 
+    exit(2)
+
+date = pin[4:6]
+month = int(pin[2:4])
+year = int(pin[:2])
+
+#print(type(month))
+
+sex = 'Muz'
+
+
+if month > 12:
+    month = month - 50
+    sex = 'Zena'
+
+century = '19'
+if year <= 22:
```

Comment:
Tohle je šikovná detekce století (s tím předpokladem, že tam nikdo nezadá RČ člověka s 100+ lety, chápu 😄) - ale když program spustím za dva roky (rok `2024`) a vložím tam RČ děcka s 1 rokem (tedy narozené _`2023`, tedy RČ s `23` na začátku)_, tak se může stát, že mi to vyplivne 19 století, že?

Čili možná tady místo _natvrdo_ zadaného čísla `22` bych to počítal podle aktuálního data (roku):

```
from datetime import datetime
current_year_full = datetime.now().year
current_century = int(str(current_year_full)[:2])
current_year = int(str(current_year_full)[2:])
year_full = (current_century if year <= current_year else current_century - 1 ) * 100 + year
```

A _(pokud jsem si to otestoval správně 😅 )_ pak v `year_full` budu mít správně celý 4-číselný rok narození daného člověka - ať už tenhle program pustím v jakémkoliv roce _(pokud rok nebude vyšší než `9999` - ale to by snad nemělo vadit, až se to někomu na pár tisíc let rozbije - a případně bude plno času na úpravu programu 🤔)_.